custom/src/main/java/org/kiwix/kiwixmobile/custom/download/CustomDownloadFragment.kt

```diff
@@ -56,24 +58,44 @@ class CustomDownloadActivity : BaseActivity() {
   @Inject lateinit var viewModelFactory: ViewModelProvider.Factory
 
   private val compositeDisposable = CompositeDisposable()
+  override fun inject(baseActivity: BaseActivity) {
+    baseActivity.customActivityComponent.inject(this)
+  }
 
-  override fun onCreate(savedInstanceState: Bundle?) {
+  override fun onCreateView(
+    inflater: LayoutInflater,
+    container: ViewGroup?,
+    savedInstanceState: Bundle?
+  ): View? {
     super.onCreate(savedInstanceState)
-    setContentView(R.layout.activity_custom_download)
-    downloadViewModel.state.observe(this, Observer(::render))
+    val root = inflater.inflate(R.layout.fragment_custom_download, container, false)
+    val activity = requireActivity() as CoreMainActivity
+    downloadViewModel.state.observe(viewLifecycleOwner, Observer(::render))
     compositeDisposable.add(
       downloadViewModel.effects.subscribe(
-        { it.invokeWith(this) },
+        { it.invokeWith(activity) },
         Throwable::printStackTrace
       )
     )
+    return root
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
     cd_download_button.setOnClickListener { downloadViewModel.actions.offer(ClickedDownload) }
     cd_retry_button.setOnClickListener { downloadViewModel.actions.offer(ClickedRetry) }
   }
 
   override fun onDestroy() {
     super.onDestroy()
     compositeDisposable.clear()
+    // findNavController().navigate(R.id.customReaderFragment)
+    findNavController().apply {
```

Comment:
we don't want any of this in onDestroy, delete it all and work inside the SideEffect